amd_openvx_extensions/amd_nn/src/tensor_matrix_multiply.cpp

```diff
@@ -83,10 +83,10 @@ static vx_status VX_CALLBACK validate(vx_node node, const vx_reference parameter
     ERROR_CHECK_STATUS(vxSetMetaFormatAttribute(metas[4], VX_TENSOR_DIMS, output_dims, sizeof(output_dims)));
 
     // check that tensors are 2D
-    if (input1_dims[3] != 1 || input1_dims[2] != 1 ||
-        input2_dims[3] != 1 || input2_dims[2] != 1 ||
-        input3_dims[3] != 1 || input3_dims[2] != 1 ||
-        output_dims[3] != 1 || output_dims[2] != 1)
+    if ((input1_dims[3] != 1 || input1_dims[2] != 1) && (input1_dims[1] != 1 || input1_dims[0] != 1) ||
+        (input2_dims[3] != 1 || input2_dims[2] != 1) && (input2_dims[1] != 1 || input2_dims[0] != 1) ||
```

Comment:
this condition is ambiguous.
Please change is something like ((input1_dims[3]&input1_dims[2]) != 1) || ((input1_dims[1]&input1_dims[0]) != 1)